src/TableColumn/KeyValue.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace atk4\ui\TableColumn;
+
+use atk4\data\Field;
+use atk4\ui\Exception;
+
+/**
+ * Class KeyValue
+ *
+ * if field have values without a relation
+ * like a status or a coded state of a process
+ * Ex :
+ * Machine state :
+ * 0 => off
+ * 1 => powerup
+ * 2 => on
+ * 3 => resetting
+ * 4 => error
+ *
+ * we don't need a table to define this, cause are defined in project
+ *
+ * using KeyValue Column you can show this values without using DB Relations
+ * need to be defined in field like this :
+ *
+ * $this->addField('course_payment_status', [
+ * 'caption' => __('Payment Status'),
+ * 'default' => 0,
+ * 'values' => [
+ * 0 => __('not invoiceable'),
+ * 1 => __('ready to invoice'),
+ * 2 => __('invoiced'),
+ * 3 => __('paid'),
+ * ],
+ * 'ui'      => [
+ * 'form' => [
+ * 'DropDown'
+ * ],
+ * 'table' => [
+ * 'KeyValue'
+ * ]
+ * ],
+ * ]);
+ *
+ *
+ * @package atk4\ui\TableColumn
+ */
+class KeyValue extends Generic
```

Comment:
I think TableColumn/Generic should have this functionality implemented.

Reply:
Ok it try to merge it